metals/src/main/scala/scala/meta/internal/metals/JvmSignatures.scala

```diff
@@ -27,10 +27,20 @@ object JvmSignatures {
           loop(tail, fqcn.append(delimiter))
       }
     }
-    val fqcn = loop(definition.symbol.ownerChain, new StringBuilder)
+    val cls = findClass(definition.symbol)
+    val fqcn = loop(cls.ownerChain, new StringBuilder)
     TypeSignature(fqcn)
   }
 
+  private def findClass(symbol: String): String = {
```

Comment:
We also take methods into account now, which in case of java files will mean that the breakpoints will work properly even with an inner class.